Examples for the manipulation of obs sequences by selecting them on a map

mjs2369 opened the pull request:
These two examples launch Dash apps, which plot observations in a sequence on a map, and allows users to draw a shape on a map to select observations in the region. 

Once obs are selected, they can either be removed from the current obs sequence or added to a new one. The resulting obs seqs are then saved in examples/04_interacting 

![image](https://github.com/user-attachments/assets/3f33c9a5-4918-45aa-a244-6c19425fe86b)

Also adds new source code dir interactive_plots

hkershaw-brown commented:
This is cool Marlee,
I have a couple of thoughts.
I don't think the examples gallery is selling this, because there is no picture. 
![Screenshot 2025-07-09 at 3 55 42 PM](https://github.com/user-attachments/assets/7f84d9c2-ae06-4825-b97d-71b51ab77c2b)

How about a youtube video of starting the app, selecting and creating a new obs seq?

I don't think refreshing the app resets the selected obs, 

<img width="984" alt="Screenshot 2025-07-09 at 2 57 57 PM" src="https://github.com/user-attachments/assets/f5471d2b-01ea-4b24-ae0c-55efa69ed62b" />

You can reproduce this if you select some obs, create a file, refresh, then select a space with no obs, you still get the previous obs.

mjs2369 commented:
@hkershaw-brown I've addressed all the comments on this, improved the organization of the spinx gallery examples, and improved the aesthetics of the Dash app. 

I think the check is failing because the interactive_plot module is not available on main, and that is what pytest is using? The tests are running just fine on my dev branch

hkershaw-brown commented:
> @hkershaw-brown I've addressed all the comments on this, improved the organization of the spinx gallery examples, and improved the aesthetics of the Dash app.
> 
> I think the check is failing because the interactive_plot module is not available on main, and that is what pytest is using? The tests are running just fine on my dev branch

I added __init__.py, and added dash to the dependencies so the test run. Not checked the code yet.

hkershaw-brown commented on examples/04_interacting/select_to_remove_obs.py:
```diff
@@ -0,0 +1,57 @@
+    app.run(debug=True, use_reloader=True)
+
+###########################################
+# Launch the Dash app by running the Python program in your terminal
+# using the ``python3 select_to_remove_obs.py`` command.
```
This seems to be missing instructions for looking at the dash app, e.g. open in a browser

hkershaw-brown commented:
@mjs2369 There still seems to be some problems with this DASH app.  
It is printing "New observation sequence created .." when I do the selection sometimes without clicking submit. 
<img width="962" height="691" alt="Screenshot 2025-08-28 at 2 42 08 PM" src="https://github.com/user-attachments/assets/1c195bdf-59e2-4029-bbe6-4d6d11014348" />

hkershaw-brown commented on src/pydartdiags/interactive_plots/interactive_plots.py:
```diff
@@ -0,0 +1,259 @@
+
+# Creates Dash app to create a new obs_seq file including only
+# the selected obs
+def include_only_selected_obs_dash_app(fig, obs_seq_selected):
```
Can users pass their own figure to this?

mjs2369 replied:
No the fig is defined in the src code function geo_plot (interactive_plots.py), which creates the geographic plot for the obs sequence that is passed in at [https://github.com/NCAR/pyDARTdiags/blob/31fa261206cbb46d3f4c2fdcf46db8ba101fd6ad/examples/04_interacting/select_to_remove_obs.py#L47](https://github.com/NCAR/pyDARTdiags/blob/31fa261206cbb46d3f4c2fdcf46db8ba101fd6ad/examples/04_interacting/select_to_remove_obs.py#L47)

This bit of code could be moved from the src code to the examples if we want our users to be able to pass their own figures in, but I think that would only be useful if we wanted them to be able to change the hover info and such.

hkershaw-brown replied:
@mjs2369 no need to respond to the comments, I closed the pull request

hkershaw-brown commented:
@mjs2369 Marlee, I'm going to close this pull request, because it seems to be buggy using dash, and slow for large numbers of obs. I think at the moment this is not something we want to support as part of this package.
